File: test/sample.cpp
// Sample C++ file for testing code extraction
#include <iostream>
#include <vector>
#include <string>

class Calculator {
private:
    std::vector<std::string> operations;

public:
    Calculator() {}

    int add(int a, int b) {
        int result = a + b;
        operations.push_back(std::to_string(a) + " + " + std::to_string(b) + " = " + std::to_string(result));
        return result;
    }

    int multiply(int a, int b) {
        int result = a * b;
        operations.push_back(std::to_string(a) + " * " + std::to_string(b) + " = " + std::to_string(result));
        return result;
    }

    std::string getHistory() {
        std::string result;
        for (size_t i = 0; i < operations.size(); ++i) {
            if (i > 0) result += "\n";
            result += operations[i];
        }
        return result;
    }
};

int main() {
    Calculator calc;
    calc.add(5, 3);
    calc.multiply(4, 2);
    std::cout << calc.getHistory() << std::endl;
    return 0;
}

File: test/sample.php
<?php
// Sample PHP file for testing code extraction

class Calculator {
    private $operations;

    public function __construct() {
        $this->operations = array();
    }

    public function add($a, $b) {
        $result = $a + $b;
        $operation = $a . " + " . $b . " = " . $result;
        array_push($this->operations, $operation);
        return $result;
    }

    public function multiply($a, $b) {
        $result = $a * $b;
        $operation = $a . " * " . $b . " = " . $result;
        array_push($this->operations, $operation);
        return $result;
    }

    public function getHistory() {
        return implode("\n", $this->operations);
    }
}

// Usage
$calc = new Calculator();
$calc->add(5, 3);
$calc->multiply(4, 2);
echo $calc->getHistory() . "\n";
?>